pennylane/tape/__init__.py

```diff
@@ -0,0 +1,90 @@
+# Copyright 2018-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This subpackage contains various quantum tapes, which track, queue,
+validate, execute, and differentiate quantum circuits.
+"""
+import contextlib
+from unittest import mock
+
+from .circuit_graph import TapeCircuitGraph
+from .measure import MeasurementProcess, state
+from .queuing import AnnotatedQueue, Queue, QueuingContext
+from .tapes import QuantumTape, QubitParamShiftTape, CVParamShiftTape, ReversibleTape
+from .qnode import QNode, qnode
+
+
+_mock_stack = []
+
+
+def enable_tape():
+    """Enable tape mode.
+
+    Tape mode is an experimental new mode of PennyLane. QNodes created in tape mode have support for
+    in-QNode classical processing, differentiable quantum decompositions, returning the quantum
+    state, less restrictive QNode signatures, and various other improvements.
+
+    For more details on tape mode, see :mod:`~.tape`.
+
+    **Example**
+
+    Simply call this function at the beginning of your script or session.
+
+    >>> qml.enable_tape()
+
+    All subsequent QNodes will be created using tape mode, and can take
+    advantage of the various tape mode features:
+
+    >>> dev = qml.device("default.qubit", wires=1)
+    >>> @qml.qnode(dev)
+    ... def circuit(x, y):
+    ...     qml.RX(np.sin(x) * y, wires=0)
+    ...     return qml.expval(qml.PauliZ(0))
+    >>> print(circuit(0.5, 0.1))
+    0.9988509758748578
+    >>> qml.grad(circuit)(0.5, 0.1)
+    (array(-0.00420574), array(-0.02297608)))
+    >>> type(circuit)
+    pennylane.tape.qnode.QNode
+
+    Tape mode can be disabled by calling :func:`~.disable_tape`.
+    """
+    if _mock_stack:
+        return

```

Comment:
(Minor suggestion:) wonder if it would be worth adding a warning here and also changing the error for `disable_tape` to be a warning?

Reply:
Good idea!

Reply:
done